src/libcore/slice/mod.rs

```diff
@@ -2146,14 +2146,13 @@ impl<T> [T] {
     /// Copying four bytes within a slice:
     ///
     /// ```
-    /// # #![feature(copy_within)]
     /// let mut bytes = *b"Hello, World!";
     ///
     /// bytes.copy_within(1..5, 8);
     ///
     /// assert_eq!(&bytes, b"Hello, Wello!");
     /// ```
-    #[unstable(feature = "copy_within", issue = "54236")]
+    #[stable(feature = "copy_within", since = "1.37.0")]
     pub fn copy_within<R: ops::RangeBounds<usize>>(&mut self, src: R, dest: usize)
     where
         T: Copy,
```

Comment:
drive-by unrelated to stabilization: seems like we shouldn't be monomorphizing this whole function on `R`, given that we only need `.start_bound()` and `.end_bound()`, which we can call first and pass into a non-generic version.

Reply:
Wouldn’t the method still be generic over `T`?

Reply:
LLVM will deduplicate identical functions anyway. Hard to see this coming at a compilation time cost and if it ever does in practice, it would be great to fix it in general rather than implementing hacks for specific functions in libcore.

Reply:
@nagisa the point is that they wouldn't be identical functions because they'd have different types for `R`. So we'd generate separate copies for `Range`, `RangeFrom`, `RangeTo`, `RangeInclusive`, etc...

Reply:
@cramertj if the generated IR after inlining is identical, it will get merged. LLVM does not care about the type system.

Reply:
My point is that I don't think the generated IR *would* be identical. `RangeFrom<T>` and `Range<T>` don't have the same representation, so converting them to start and end bounds would need to be split out into a separate function in order for the inner body of the functions to be unified.